praktikumsplaner-frontend/frontend/src/components/Meldezeitraeume/CreateMeldezeitraum.vue

```diff
@@ -0,0 +1,147 @@
+                                        ></v-text-field>
+                                    </v-col>
+                                </v-row>
+                                <v-row>
+                                    <v-col cols="12">
+                                        <ZeitraumPicker
+                                            :value="zeitraum"
+                                            @input="zeitraumChange"
+                                        ></ZeitraumPicker>
+                                    </v-col>
+                                </v-row>
+                            </v-container>
+                        </v-card-text>
+                    </v-form>
+                </v-card>
+            </v-dialog>
+        </v-row>
+    </v-container>
+</template>
+<script setup lang="ts">
+import { useSnackbarStore } from "@/stores/snackbar";
+import { Levels } from "@/api/error";
+import { ref } from "vue";
+import Meldezeitraum from "@/types/Meldezeitraum";
+import MeldezeitraumService from "@/api/MeldezeitraumService";
+import { useRules } from "@/composables/rules";
+import ZeitraumPicker from "@/components/Meldezeitraeume/ZeitraumPicker.vue";
+import Zeitraum from "@/types/Zeitraum";
+
+const meldezeitraum = ref<Meldezeitraum>(new Meldezeitraum(""));
+const zeitraum = ref<Zeitraum>(new Zeitraum());
+
+const form = ref<HTMLFormElement>();
+const dialog = ref<boolean>(false);
+const meldezeitraumService = new MeldezeitraumService();
+const maxLength = 255;
+const validationRules = useRules();
+
+const zeitraumNameRules = [
+    validationRules.maxLengthRule(
+        maxLength,
+        "Der Name darf maximal " + maxLength + " Zeichen lang sein."
+    ),
+    validationRules.notEmptyRule("Der Zeitraumname darf nicht leer sein."),
+];
+
+const emits = defineEmits<{
+    (e: "meldezeitraumAdded", meldezeitraum: Meldezeitraum): void;
+}>();
+
+function zeitraumChange(changedZeitraum: Zeitraum) {
+    zeitraum.value = changedZeitraum;
+    meldezeitraum.value.startZeitpunkt = changedZeitraum.startZeitpunkt;
+    meldezeitraum.value.endZeitpunkt = changedZeitraum.endZeitpunkt;
+}
+
+function resetForm() {
+    meldezeitraum.value = new Meldezeitraum("");
+    zeitraum.value = new Zeitraum();
+    form.value?.resetValidation();
+}
+
+function clickSpeichern() {
+    if (form.value?.validate()) {
+        dialog.value = false;
+        meldezeitraumService
+            .create(meldezeitraum.value)
+            .then(() => {
+                useSnackbarStore().showMessage({
+                    message: "Meldezeitraum erfolgreich angelegt",
+                    level: Levels.INFO,
+                });
+                emits("meldezeitraumAdded", meldezeitraum.value);
+            })
+            .catch((error) => {
+                useSnackbarStore().showMessage({
+                    message: error.message,
+                    level: Levels.ERROR,
+                });
```

Comment:
these should be part of the service because the feedback should normally always appear after failure, who calls it.